### DISCUSSION TITLE
[QST] rmm/cuda_stream_view.hpp:21:10: fatal error: cuda/stream_ref: No such file or directory

### DISCUSSION THREAD

**Discussion by hsgui** [rapidsai/rmm#1647] [Q&A] [answered]
**What is your question?**

**environment:**

- cuda SDK: 12.1
- compute_cap: 7.5

But I got compile error when compiling `rmm/cuda_stream_view.hpp`

`/data/workspace/conda/envs/cuvs/include/rmm/cuda_stream_view.hpp:21:10: fatal error: cuda/stream_ref: No such file or directory
   21 | #include <cuda/stream_ref>
      |
`

I also checked path `/usr/local/cuda/include/cuda`, there is no `stream_ref` in it.

could anyone help me fix the compile issue?

thanks!

**Comment by harrism** [comment]
Hi @hsgui, can you provide details of how you are compiling?

**Comment by hsgui** [comment]
> Hi @hsgui, can you provide details of how you are compiling?

```bash
/usr/local/gcc12/bin/g++ cuvs_hello.cc -o cuvs_hello   -I /usr/local/cuda/include/   -I /data/workspace/conda/envs/cuvs/include   -L /data/workspace/conda/envs/cuvs/lib/   -L /usr/local/cuda/lib64/   -DSPDLOG_FMT_EXTERNAL=1
```

besides, I'm building cuvs from C++ source file from this wiki: 

`https://docs.rapids.ai/api/cuvs/stable/build/#build-from-source`

**Comment by hsgui** [comment]
> Hi @hsgui, can you provide details of how you are compiling?

I think i need upgrade the cuda toolkit maybe?

the `stream_ref` is included within `https://github.com/NVIDIA/libcudacxx/commit/e9c40bd732d05160651d19ae1d8b4f3c10bb890b`, which was committed in **Mar 6, 2023**.

but my `nvcc --version` shows following:

```bash
[root@xxx cuda-demo]# nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Feb__7_19:32:13_PST_2023
Cuda compilation tools, release 12.1, V12.1.66
Build cuda_12.1.r12.1/compiler.32415258_0
```

**Comment by harrism** [comment]
Using a more recent CUDA toolkit is one option, but not the option RAPIDS relies on. We use cmake to fetch CCCL from GitHub.

Building manually you will need to provide a -I line to point to CCCL include directory.

Using cmake will save you a lot of trouble. You could base it off of the cmakelists.txt from the RMM tests or benchmarks.

**Comment by hsgui** [comment]
thanks very much @harrism 

I'll play with cmake

**Comment by harrism** [comment, marked as answer]
I'm going to convert this to a discussion. You can "Mark as answer" a comment that helps if / when you are satisfied. Thanks!
